subsys/bluetooth/host/gatt.c

```diff
@@ -2140,18 +2180,19 @@ static int gatt_notify_mult(struct bt_conn *conn, uint16_t handle,
 	}
 
 	BT_DBG("handle 0x%04x len %u", handle, params->len);
+	gatt_add_nfy_to_buf(*buf, handle, params);
 
-	nfy = net_buf_add(*buf, sizeof(*nfy));
-	nfy->handle = sys_cpu_to_le16(handle);
-	nfy->len = sys_cpu_to_le16(params->len);
-
-	net_buf_add(*buf, params->len);
-	memcpy(nfy->value, params->data, params->len);
+	/* Increment the number of handles, ensuring the notify callback gets
+	 * called once for every attribute.
+	 */
+	bt_att_increment_tx_meta_data_attr_count(*buf, 1);
 
-	k_work_submit(&nfy_mult_work);
+	k_work_schedule(&nfy_mult_work,
+			K_MSEC(CONFIG_BT_GATT_NOTIFY_MULTIPLE_FLUSH_MS));
```

Comment:
Can we end up here while the `nfy_mult_work` has already been scheduled? 

If so, should we reschedule it, or keep the original schedule?

Reply:
yes, this can happen (and does if using a long flush timeout). I think it's good to reschedule it for power-saving, but maybe it's not clear enough to the user. @alwa-nordic wanted to write some more documentation for the gatt notify APIs, he can mention this behavior.

Reply:
I guess it'll end up being a power vs time optimization. If we reschedule with e.g. 1ms delay each time, sending e.g. 5 notifications may be delayed up to 5ms (but I guess that's still low compared to typical connection interval).

Reply:
We should not reschedule. The API I have in mind says "A notification posted with `bt_gatt_notify_by` will be delayed at most <configured max delay>".

I believe it is preferable to an API that flushes after a period without activity. Consider: If an applications trickles notifications, those should be delivered in a timely fashion, and not batched.

Reply:
Agreed; let's not reschedule (may add a comment in the code as to why we do `k_work_schedule` instead of `k_work_reschedule`)

Reply:
```suggestion
	/* `k_work_schedule` will not reschedule. This is good because we want the oldest
	 * notification in the batch to determine the timeout, so that there is an upper bound
	 * on the delay of each notification.
	 */
	k_work_schedule(&nfy_mult_work,
			K_MSEC(CONFIG_BT_GATT_NOTIFY_MULTIPLE_FLUSH_MS));
```